assets/hip-206/solidity/TokenCreateContract.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.6.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "./HederaTokenService.sol";
+import "./IHederaTokenService.sol";
+import "./HederaResponseCodes.sol";
+
+contract TokenCreateContract is HederaTokenService {
+
+    using Bits for uint;
+
+    // Create Fungible Token with no custom fees, with a user account as admin, contract as supply and pause key.
+    function createFungible(address contractKey) external returns (address createdTokenAddress) {
+
+        // create TokenKey of type adminKey with value inherited from called account
+        uint adminKeyType;
+        IHederaTokenService.KeyValue memory adminKeyValue;
+        // turn on bit corresponding to admin key type
+        adminKeyType = adminKeyType.setBit(0);
+        // set the value of the key to be inherited from the calling account
+        adminKeyValue.inheritAccountKey = true;
+        
+        // create TokenKey of types supplyKey and pauseKey with value a contract address passed as function arg
+        uint supplyPauseKeyType;
+        IHederaTokenService.KeyValue memory supplyPauseKeyValue;
+        // turn on bits corresponding to supply and pause key types
+        supplyPauseKeyType = supplyPauseKeyType.setBit(4);
```

Comment:
We should have constants for these in IHederaTokenService instead of relying on magic numbers.